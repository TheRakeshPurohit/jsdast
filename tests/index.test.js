const unified = require('unified')
const toVFile = require('to-vfile')

const jsdastParser = require('..')

test('basic from vfile', () => {
  const tree = unified().use(jsdastParser).parse(toVFile('./example/index.js'))

  for (const node of tree.children) {
    delete node.path
  }

  expect(tree).toMatchSnapshot()
})

test('basic from plain content', () => {
  const tree = unified().use(jsdastParser).parse(`
/**
 * @param {number} a
 * @param {number} b
 * @returns {number}
 */
function sum(a, b) {
  return a + b
}
  `)

  for (const node of tree.children) {
    delete node.path
  }

  expect(tree).toMatchSnapshot()
})
