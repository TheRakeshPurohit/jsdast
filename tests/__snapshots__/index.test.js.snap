// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic from plain content 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "doc": Object {
            "description": "",
            "tags": Array [
              Object {
                "name": "a",
                "tagName": "param",
                "text": "a",
                "typeExpression": "number",
              },
              Object {
                "name": "b",
                "tagName": "param",
                "text": "b",
                "typeExpression": "number",
              },
              Object {
                "name": undefined,
                "tagName": "returns",
                "text": undefined,
                "typeExpression": "number",
              },
            ],
          },
          "isDefaultExport": false,
          "isExported": false,
          "name": "sum",
          "parameters": Array [
            Object {
              "initializer": undefined,
              "isReadOnly": false,
              "isRestParameter": false,
              "name": "a",
              "type": "Parameter",
              "valueType": "number",
            },
            Object {
              "initializer": undefined,
              "isReadOnly": false,
              "isRestParameter": false,
              "name": "b",
              "type": "Parameter",
              "valueType": "number",
            },
          ],
          "type": "FunctionDeclaration",
          "valueType": "number",
        },
      ],
      "doc": Object {
        "description": "",
        "tags": Array [],
      },
      "name": "index.js",
      "type": "Module",
    },
  ],
  "type": "Root",
}
`;

exports[`basic from vfile 1`] = `
Object {
  "children": Array [],
  "type": "Root",
}
`;
